src/main/java/core/basesyntax/service/parse/DataParserImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.parse;
+
+import core.basesyntax.model.FruitRecord;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataParserImpl implements DataParser {
+
+    @Override
+    public List<FruitRecord> parseFruitRecords(List<String> lines) {
+        List<FruitRecord> transactions = new ArrayList<>();
+
+        for (int i = 1; i < lines.size(); i++) {
+            String line = lines.get(i);
+            String[] parts = line.split(",");
+
+            String operationCode = parts[0].trim();
+            String fruit = parts[1].trim();
+            int quantity = Integer.parseInt(parts[2].trim());
+
+            FruitRecord.Operation operation = FruitRecord.Operation.fromCode(operationCode);
+            transactions.add(new FruitRecord(operation, fruit, quantity));
+        }
```

Comment:
This parsing method doesn't handle potential exceptions that may occur due to incorrect data format. Consider adding validation for the input data and throw a RuntimeException with an informative message if the data is invalid. For example, if the parts array doesn't have the expected length or if the quantity is not a valid integer, you should handle these cases appropriately.